syncto/tests/test_authentication.py

```diff
@@ -57,8 +58,39 @@ def test_should_cache_credentials_the_second_time(self):
                 build_sync_client(self.request)
                 SyncClient.assert_called_with(**self.credentials)
                 # Second time
-                self.request.headers[CLIENT_STATE_HEADER] = '6789'
                 build_sync_client(self.request)
                 # TokenServerClient should have been called only once.
                 TSClient.assert_called_once_with('1234', '12345')
                 SyncClient.assert_called_with(**self.credentials)
+
+    def test_should_cache_encrypted_credentials(self):
+        self.request.headers = {AUTHORIZATION_HEADER: 'Browserid 1234',
+                                CLIENT_STATE_HEADER: '12345'}
+        self.request.registry.cache.flush()
+        with mock.patch('syncto.authentication.TokenserverClient') as TSClient:
+            TSClient.return_value.get_hawk_credentials.return_value = \
+                self.credentials
+            with mock.patch('syncto.authentication.encrypt',
+                            return_value='encrypted'):
+                with mock.patch.object(self.request.registry.cache, 'set') \
+                        as mocked_set:
+                    build_sync_client(self.request)
+                    cache_key = ('credentials_636130e072155efd00d8e27196500'
+                                 'd29110fb2e8e93bcedb2a30e0aa0e5ccf61')
+                    mocked_set.assert_called_with(cache_key,
+                                                  'encrypted', 300)
+
+    def test_should_decrypt_credentials(self):
+        self.request.headers = {AUTHORIZATION_HEADER: 'Browserid 1234',
+                                CLIENT_STATE_HEADER: '12345'}
+        with mock.patch.object(
+                self.request.registry.cache, 'get',
+                return_value=ENCRYPTED_CREDENTIALS):
+            with mock.patch('requests.request'):
+                with mock.patch('syncto.authentication.decrypt',
+                                return_value=json.dumps(self.credentials)) \
+                        as mocked_decrypt:
+                    build_sync_client(self.request)
+                    mocked_decrypt.assert_called_with(ENCRYPTED_CREDENTIALS,
+                                                      '12345',
+                                                      'This is not a secret')
```

Comment:
I prefer to raise and not drop the value form the cache (because it could be used to invalidate cache.)
If it raise it means something went wrong so we should probably raise in that case.